cmake/helpers/CheckDependentLibraries.cmake

```diff
@@ -173,114 +186,79 @@ endif ()
                    CAN_DISABLE)
 
 gdal_check_package(GeoTIFF "libgeotiff library (external)" CAN_DISABLE)
-if (NOT GDAL_USE_GEOTIFF)
-  set(GDAL_USE_LIBGEOTIFF_INTERNAL
-      ON
-      CACHE BOOL "Use internal libgeotiff copy (if set to ON, has precedence over GDAL_USE_GEOTIFF)")
-  if (NOT GDAL_USE_LIBGEOTIFF_INTERNAL)
-    message(FATAL_ERROR "GDAL_USE_GEOTIFF or GDAL_USE_LIBGEOTIFF_INTERNAL must be set to ON")
-  endif ()
-else ()
-  set(GDAL_USE_LIBGEOTIFF_INTERNAL
-      OFF
-      CACHE BOOL "Use internal libgeotiff copy (if set to ON, has precedence over GDAL_USE_GEOTIFF)")
+invert_on_off(GDAL_USE_GEOTIFF NOT_GDAL_USE_GEOTIFF)
+set(GDAL_USE_LIBGEOTIFF_INTERNAL
+    ${NOT_GDAL_USE_GEOTIFF}
+    CACHE BOOL "Use internal libgeotiff copy (if set to ON, has precedence over GDAL_USE_GEOTIFF)")
+if ((NOT GDAL_USE_GEOTIFF) AND (NOT GDAL_USE_LIBGEOTIFF_INTERNAL))
+  message(FATAL_ERROR "GDAL_USE_GEOTIFF or GDAL_USE_LIBGEOTIFF_INTERNAL must be set to ON")
 endif ()
 
 gdal_check_package(PNG "PNG compression library (external)" CAN_DISABLE)
-if (NOT GDAL_USE_PNG)
-  set(GDAL_USE_LIBPNG_INTERNAL
-      ON
-      CACHE BOOL "Use internal libpng copy (if set to ON, has precedence over GDAL_USE_PNG)")
-else ()
-  set(GDAL_USE_LIBPNG_INTERNAL
-      OFF
-      CACHE BOOL "Use internal libpng copy (if set to ON, has precedence over GDAL_USE_PNG)")
-endif ()
+invert_on_off(GDAL_USE_PNG NOT_GDAL_USE_PNG)
+set(GDAL_USE_LIBPNG_INTERNAL
+    ${NOT_GDAL_USE_PNG}
+    CACHE BOOL "Use internal libpng copy (if set to ON, has precedence over GDAL_USE_PNG)")
 
 gdal_check_package(JPEG "JPEG compression library (external)" CAN_DISABLE)
-if (NOT GDAL_USE_JPEG)
-  set(GDAL_USE_LIBJPEG_INTERNAL
-      ON
-      CACHE BOOL "Use internal libjpeg copy (if set to ON, has precedence over GDAL_USE_JPEG)")
-else ()
-  set(GDAL_USE_LIBJPEG_INTERNAL
-      OFF
-      CACHE BOOL "Use internal libjpeg copy (if set to ON, has precedence over GDAL_USE_JPEG)")
+if (GDAL_USE_JPEG AND (JPEG_LIBRARY MATCHES ".*turbojpeg\.(so|lib)"))
+  message(
+    FATAL_ERROR
+      "JPEG_LIBRARY should point to a library with libjpeg ABI, not TurboJPEG. See https://libjpeg-turbo.org/About/TurboJPEG for the difference"
+    )
 endif ()
+invert_on_off(GDAL_USE_JPEG NOT_GDAL_USE_JPEG)
+set(GDAL_USE_LIBJPEG_INTERNAL
+    ${NOT_GDAL_USE_JPEG}
+    CACHE BOOL "Use internal libjpeg copy (if set to ON, has precedence over GDAL_USE_JPEG)")
 
 gdal_check_package(GIF "GIF compression library (external)" CAN_DISABLE)
-if (NOT GDAL_USE_GIF)
-  set(GDAL_USE_GIFLIB_INTERNAL
-      ON
-      CACHE BOOL "Use internal giflib copy (if set to ON, has precedence over GDAL_USE_GIF)")
-else ()
-  set(GDAL_USE_GIFLIB_INTERNAL
-      OFF
-      CACHE BOOL "Use internal giflib copy (if set to ON, has precedence over GDAL_USE_GIF)")
-endif ()
+invert_on_off(GDAL_USE_GIF NOT_GDAL_USE_GIF)
+set(GDAL_USE_GIFLIB_INTERNAL
```

Comment:
I would prefer if the variable was named `GDAL_USE_GIF_INTERNAL`, similar to `GDAL_USE_GIF`.

Reply:
done